python/tvm/relax/transform/schedule.py

```diff
@@ -0,0 +1,109 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# pylint: disable=invalid-name, unused-argument
+"""Relax passes related to scheduling functions for target hardware."""
+import tempfile
+from typing import Union, List
+
+from tvm import relax
+from tvm.ir import transform, IRModule
+from tvm.target import Target
+from tvm import meta_schedule as ms
+from .tuning_api import Trace
+
+
+@transform.module_pass(opt_level=2, name="schedule_for_target")
+class ScheduleForTarget:
+    """Apply a minimal set of transformations to enable running on a specific target."""
+
+    def __init__(self, target: Union[Target, str], trials_per_task: int = 4):
+        """
+        This function returns a pass which applies basic schedule transformations to each
+        primitive function in the input module for the specified target. This is useful
+        when a hardware target requires certain intrinsics for kernels to be valid. For
+        example, on GPUs, each kernel must have loops bound to a thread and block index.
+        By default, primitive functions do not contain this binding. Applying a single
+        step of Metaschedule's transform rules inserts bindings that enable the functions
+        to be run.
+
+        Thus, this pass is a convenience wrapper around the simplist possible invocation
+        of Metaschedule tuning. It performs only a few schedules per task, skips benchmarking,
+        and verifies that they can be built.
+
+        Parameters
+        ----------
+        target : Union[Target, str]
+            The tvm target that fucntions should be scheduled for.
+        trials_per_task : int
```

Comment:
Just trying to understand what's the relationship between `trials_per_task` and `max_trial_per_task` and should we unify them?